crates/svm-cli/tests/spawn_app_tests.rs

```diff
@@ -0,0 +1,216 @@
+use regex::Regex;
+
+use svm_cli::{cli, wasm_value};
+use svm_common::Address;
+
+struct SpawnAppTestCase {
+    version: String,
+    template_addr: String,
+    ctor_index: String,
+    ctor_buf: String,
+    ctor_args: Vec<String>,
+}
+
+#[test]
+fn encode_decode() {
+    let cases = vec![
+        SpawnAppTestCase {
+            version: String::from("0"),
+            template_addr: String::from("00aa00aa00aa00aa00aa00aa00aa00aa00aa00aa"),
+            ctor_index: String::from("0"),
+            ctor_buf: String::from("11bb11bb12345678"),
+            ctor_args: vec![String::from("10i32"), String::from("20i64")],
+        },
+        SpawnAppTestCase {
+            version: String::from("0"),
+            template_addr: String::from("00aa00aa00aa00aa00aa00aa00aa00aa00aa00aa"),
+            ctor_index: String::from("12"),
+            ctor_buf: String::from(""),
+            ctor_args: vec![String::from("1073741824i64"), String::from("0i32")],
+        },
+    ];
+
+    for case in cases {
+        test_encode_decode(case)
+    }
+}
+
+fn test_encode_decode(case: SpawnAppTestCase) {
+    let tempfile_path = tempfile::NamedTempFile::new().unwrap();
+    let tempfile_path = tempfile_path.path().to_str().unwrap();
+
+    let output_path = tempfile_path;
+    let mut input = vec![
+        "myprog",
+        "encode",
+        "spawn_app",
+        output_path,
+        &case.version,
+        &case.template_addr,
+        &case.ctor_index,
+        &case.ctor_buf,
+    ];
+    for i in 0..case.ctor_args.len() {
```

Comment:
It seems the loop could be replaced with:

```rust
input.extend_from_slice(&case.ctor_args[..]);
```

Reply:
Due to the `String`/`&str` conversion, the only kind of solution I found for this is:
```rust
let ctor_args: Vec<_> = case.ctor_args.iter().map(|s| s.as_str()).collect();
input.extend_from_slice(&ctor_args);
```
Which I think is more complicated IMO.
Let me know what you think.

Reply:
I've overlooked the `ctor_arg` contains `&str` (and not `String`).
AFAIK you suggestion is more idiomatic than writing a loop.